core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/imagecatalog/ImagesToImagesV4ResponseConverter.java

```diff
@@ -38,47 +47,73 @@
-            hdfImages.add(hdfImgJson);
-        }
+        res.setHdpImages(hdpImages);
         res.setHdfImages(hdfImages);
+        res.setCdhImages(cdhImages);
         res.setSupportedVersions(source.getSuppertedVersions());
         return res;
     }
 
+    private List<ImageV4Response> setBaseImages(List<Image> source) {
+        List<ImageV4Response> images = new ArrayList<>();
+        for (Image img : source) {
+            ImageV4Response imgJson = new ImageV4Response();
+            copyImageFieldsToJson(img, imgJson);
+            imgJson.setStackDetails(convertStackDetailsToJson(img.getStackDetails(), img.getOsType()));
+            images.add(imgJson);
+        }
+        return images;
+    }
+
     private List<BaseImageV4Response> getBaseImageResponses(Images source) {
-        List<StackDetailsV4Response> defaultHdpStacks = getDefaultStackInfos(defaultHDPEntries.getEntries().values());
-        List<StackDetailsV4Response> defaultHdfStacks = getDefaultStackInfos(defaultHDFEntries.getEntries().values());
+        List<AmbariStackDetailsV4Response> defaultHdpStacks = getDefaultStackInfos(defaultHDPEntries.getEntries().values());
+        List<AmbariStackDetailsV4Response> defaultHdfStacks = getDefaultStackInfos(defaultHDFEntries.getEntries().values());
+        List<ClouderaManagerStackDetailsV4Response> defaultCdhStacks = getDefaultCdhStackInfo(defaultCDHEntries.getEntries().values());
         List<BaseImageV4Response> baseImages = source.getBaseImages().stream()
-                .filter(image -> defaultAmbariRepoService.getDefault(image.getOsType()) != null)
+                .filter(image ->
+                        defaultAmbariRepoService.getDefault(image.getOsType()) != null
+                                || defaultClouderaManagerRepoService.getDefault(image.getOsType()) != null)
                 .map(image -> {
                     BaseImageV4Response imgJson = new BaseImageV4Response();
                     copyImageFieldsToJson(image, imgJson);
                     imgJson.setHdpStacks(defaultHdpStacks);
                     imgJson.setHdfStacks(defaultHdfStacks);
+                    imgJson.setCdhStacks(defaultCdhStacks);
+                    ClouderaManagerRepo clouderaManagerRepo = defaultClouderaManagerRepoService.getDefault(image.getOsType());
                     AmbariRepo ambariRepo = defaultAmbariRepoService.getDefault(image.getOsType());
+                    if (ambariRepo != null) {
+                        AmbariRepositoryV4Response ambariRepoJson = new AmbariRepositoryV4Response();
+                        ambariRepoJson.setBaseUrl(ambariRepo.getBaseUrl());
+                        ambariRepoJson.setVersion(ambariRepo.getVersion());
+                        ambariRepoJson.setGpgKeyUrl(ambariRepo.getGpgKeyUrl());
+                        imgJson.setAmbariRepo(ambariRepoJson);
+                    }
+                    if (clouderaManagerRepo != null) {
```

Comment:
Maybe an `else if` would be better to avoid misunderstanding configure both of them situation.